README.md

```diff
@@ -82,132 +94,451 @@ integration_types:
+all webhook endpoints are publically visible, but the framework has
+two options for stricter access controls.
+
+* `ANY_PARTY` - Requests must be presented with a valid Daml Hub JWT corresponding to the integration's ledger.
+* `INTEGRATION_PARTY` - Requests must be presented with a valid Daml Hub JWT corresponding to the integration's ledger and party.
+
+A JWT is considered to be valid for a given party, only if that party
+is listed in both the `readAs` and `actAs` claims.
+
+`ANY_PARTY` is intended to be used in scenarios where the integration
+might wish to enforce its own access controls based on an
+authenticated user identity. To support this, the framework has two
+functions for extracting the user's identify from an inbound
+request. Note that these functions do not return results on `PUBLIC`
+endpoints, due to the fact that there is no authentication checking
+done for these endpoints and no notion of request identity.
+
+```python
+from daml_dit_if.api import
+    get_request_parties, \
+    get_single_request_party
+
+    ... elided. ..
+
+    @events.webhook.get(label='CSV Table', auth=AuthorizationLevel.INTEGRATION_PARTY)
+    async def on_get_table_csv(request):
+        row_data = get_formatted_table_data()
+
+        LOG.info('>>> %r/%r', get_request_parties(request), get_single_request_party(request))
+```
+
+### Timer Events
+
+To support time-based activities (polling, etc.), the integration
+framework provides support for periodic timer events. These are events
+that the framework schedules to be invoked at a repeating schedule at
+a fixed interval. The interval is specified in seconds and the
+decorator contains an optional label argument used to populate the
+descriptive text on the integration status display.  As with other
+event handlers, the handler for timer events can return a list of
+ledger commands to be issued by the framework.
+
+```python
+    @events.time.periodic_interval(env.interval, label='Periodic Timer')
+    async def interval_timer_elapsed():
+        LOG.debug('Timer elapsed: %r', active_cids)
+        return [exercise(cid, env.templateChoice, {})
+                for cid
+                in active_cids]
+```
+
+The integration framework makes a best effort attempt to call the
+timer event handler at the requested periodicity, but no guarantees
+are made about specific timing. The requested periodicity should be
+considered a minimum. The event handler for a given timer will not be
+called re-entrantly.
+
+### Queue Events
+
+The integration framework also provides in-memory queues and will call
+queue event handlers for message placed on those queues. The intent of
+this capability is to provide a way for integrations to respond to
+types of events that the other event handlers do not cover.  An
+example of this is the the Symphony intgration, which uses queue
+events to accept and process incoming messages received over a
+websocket. Because there is no websocket event hander type, the socket
```

Comment:
This isn’t entirely accurate - the Symphony integration uses event listeners from Symphony’s Python API client library (which I believe uses polling). 

For the purposes of this doc however, the reasoning behind using the integration’s queue events are the same as if it was websockets directly. The Exberry integration does use websockets with the integration queue.

Reply:
Good catch. I've reworded. Thanks!